include/ringmesh/basic/frame.h

```diff
@@ -147,11 +147,79 @@ namespace RINGMesh
             return origin_;
         }
 
+        vecn< DIMENSION > coords_to_local(
+            const vecn< DIMENSION >& global_coords ) const;
+
+        vecn< DIMENSION > coords_to_global(
+            const vecn< DIMENSION >& local_coords ) const
+        {
+            vecn< DIMENSION > global_coords = ( *this ).origin();
+            for( auto coord : RINGMesh::range( DIMENSION ) )
+            {
+                for( auto coor : RINGMesh::range( DIMENSION ) )
+                {
+                    global_coords[coord] +=
+                        local_coords[coor] * ( *this )[coor][coord];
```

Comment:
You still do not prefer this->?

Reply:
should i write this->[coor][coord] then ?

Reply:
If I well remember this->[x] is not accepted by some compilators

Reply:
No, you are right for this line but I saw other lines where you could do it.